packages/next/next-server/server/image-optimizer.ts

```diff
@@ -112,7 +112,7 @@ export async function imageOptimizer(
     return { finished: true }
   }
 
-  const { href } = absoluteUrl
+  const href = (absoluteUrl || relativeUrl) as string
```

Comment:
```suggestion
  const href = absoluteUrl || relativeUrl
```

If you avoid using `href` below then you don't need this type cast